crates/forge/src/lib.rs

```diff
@@ -127,19 +127,29 @@ pub async fn run(
 
     let mut summaries = vec![];
 
-    for compiled_test_crate in test_crates {
+    for compiled_test_crate_raw in test_crates {
         pretty_printing::print_running_tests(
-            compiled_test_crate.tests_location,
-            compiled_test_crate.test_cases.len(),
+            compiled_test_crate_raw.tests_location,
+            compiled_test_crate_raw.test_cases.len(),
+        );
+
+        let maybe_test_program_path = Arc::new(
+            test_program_paths_map
+                .get(&compiled_test_crate_raw.tests_location)
+                .cloned(),
```

Comment:
I prefer to pass `Option` here explicitly and then use `Default` at the end instead of hiding the fact that the path does not exist. I have to pass sth around since we always build `ProfilerCallTrace`, even when we don't need it (it was introduced long time ago by @MaksymilianDemitraszek I believe)

Reply:
This should be refactored then somehow

Reply:
True, but not in this PR. It was prob done to avoid relying on `Option` - to make sure its typesafe we would have to rewrite the whole type system responsible for building test summaries, preferably using type families

Reply:
https://github.com/foundry-rs/starknet-foundry/issues/2068